docs/reference/tutorials-elasticsearch-customise-query.md

```diff
@@ -9,32 +9,68 @@ Elasticsearch connector allows you to customise the Elasticsearch request body b
 
 This is an advanced option, the underlying query may change between versions and reading from / mutating the query is brittle, so please be aware to use this sparingly and let us know what you want to achieve through github issues.
 
-Example below is overriding the `query` section of the Elasticsearch request body.
+## Using beforeSearchCall Hook
+
+The `beforeSearchCall` hook allows you to modify the entire request body before it's sent to Elasticsearch. This is useful when you need to modify multiple parts of the request or add custom fields.
 
 ```js
-const connector = new ElasticsearchAPIConnector(
-  {
-    host: "https://example-host.es.us-central1.gcp.cloud.es.io:9243",
-    index: "national-parks",
-    apiKey: "exampleApiKey"
-  },
-  (requestBody, requestState, queryConfig) => {
-    console.log("postProcess requestBody Call", requestBody); // logging out the requestBody before sending to Elasticsearch
-    if (!requestState.searchTerm) return requestBody;
+const connector = new ElasticsearchAPIConnector({
+  host: "https://example-host.es.us-central1.gcp.cloud.es.io:9243",
+  index: "national-parks",
+  apiKey: "exampleApiKey",
+  beforeSearchCall: async (
+    { requestBody, requestState, queryConfig },
+    next
```

Comment:
was there a reason why you needed to pass a handler to return the body? Could it just be a function return?